### ISSUE TITLE
Obtain batched embeddings

### ISSUE THREAD

**Issue by vinid** [ART-Group-it/KERMIT#1]
Hello! :) thank you for this awesome package!

I'd like to use KERMIT embeddings into another network, but I'd also like to avoid to generate all of them beforehand (to avoid high RAM usage). Thus, I'd like to pass batches of sentences in parenthetical format to the model and get a batch of embeddings in output.

It is not clear to me if one of these elements can take more then 1 sample in input (i.e., more trees/batches of trees).

``` python
tree = Tree(string=your_parse_tree)
kermit_encoder = DT(dimension=8192, LAMBDA= 0.6, operation=fast_shuffled_convolution)

kermit_tree_encoded = kermit_encoder.dt(tree)
```

Thank you very much :)

**Comment by teelinsan** [comment]
Hi @vinid,
unfortunately the current implementation does not allow to generate batched KERMIT embeddings.
Since these embeddings are not trained, usually in our workflow we generate the embeddings and then save them on a file (you can use multiprocessing to save time). During the training procedure, we load the _batch of embeddings_ on demand from the file.
If there is enough interest we can work on a next release of the library that includes batched embeddings, however, note that this won't avoid high RAM usage because the system still needs to calculate these embeddings on the fly.

To avoid high RAM usage you can try to use [Iterable-style datasets](https://pytorch.org/docs/stable/data.html#iterable-style-datasets) instead of Map-style datasets. You can also try to use a smaller batch that fits in RAM and then use gradient accumulation [(here an example in pytorch-lightning)](https://pytorch-lightning.readthedocs.io/en/stable/training_tricks.html#accumulate-gradients).

Hope this helps!

**Comment by vinid** [comment]
Hi @teelinsan 

thanks a lot! Yea my idea was to pass batches of sentences to the model during training to keep the implementation consistent with other stuff I already had in place. I think your suggestion of the Iterable-style dataset makes sense.

I'll first generate the embeddings and then load them in batches.

Thank you :)
